core/meterer/offchain_store.go

```diff
@@ -0,0 +1,239 @@
+package meterer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	commonaws "github.com/Layr-Labs/eigenda/common/aws"
+	commondynamodb "github.com/Layr-Labs/eigenda/common/aws/dynamodb"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+)
+
+type OffchainStore struct {
+	dynamoClient         *commondynamodb.Client
+	reservationTableName string
+	onDemandTableName    string
+	globalBinTableName   string
+	logger               logging.Logger
+	// TODO: add maximum storage for both tables
+}
+
+func NewOffchainStore(
+	cfg commonaws.ClientConfig,
+	reservationTableName string,
+	onDemandTableName string,
+	globalBinTableName string,
+	logger logging.Logger,
+) (*OffchainStore, error) {
+
+	dynamoClient, err := commondynamodb.NewClient(cfg, logger)
+	if err != nil {
+		return nil, err
+	}
+	if reservationTableName == "" || onDemandTableName == "" || globalBinTableName == "" {
+		return nil, fmt.Errorf("table names cannot be empty")
+	}
+
+	err = CreateReservationTable(cfg, reservationTableName)
```

Comment:
Tables should be created outside the application code (devops). We can check if the tables with corresponding names exist in the constructor and error out if they don't exist

Reply:
makes sense, updated to use check and error instead of making db changes